### DISCUSSION TITLE
Move Socket.io to `gatsby develop` port

### DISCUSSION THREAD

**Discussion by aaronadamsCA** [gatsbyjs/gatsby#31898] [Ideas / Feature Requests]
It would be preferable for Socket.io to run on the same port as `gatsby develop`, not on a separate port.

Today, any environment that depends on port forwarding breaks the separate status port, for example:

- **GitHub Codespaces** and other containerized development environments.
  - Current workaround: set the unsupported `INTERNAL_STATUS_PORT` environment variable, then manually configure port forwarding.
- **[Ngrok](https://ngrok.com/)**, which only lets you tunnel one local port to an HTTPS Internet hostname. Internet hostnames are required for some dev scenarios, like OAuth.
  - No available workaround.

#28443 seemed to suggest the effort here isn't too high, so I hope it's something you'll consider.

**Comment by ZeldOcarina** [comment]
Hi! Can you please expand on `INTERNAL_STATUS_PORT`?
